react.js

```diff
@@ -1,34 +1,58 @@
 module.exports = {
   'extends': [
-    './index.js'
+    './index.js',
     'plugin:react/recommended',
   ],
 
   'plugins': [
-    'react'
+    'react',
   ],
 
   'parserOptions': {
     'ecmaFeatures': {
-      'jsx': true
-    }
+      'jsx': true,
+    },
   },
 
   'rules': {
-    'react/forbid-prop-types': 0,
-    'react/jsx-boolean-value': [2, 'never'],
-    'react/jsx-closing-bracket-location': [1, 'tag-aligned'],
-    'react/jsx-curly-spacing': [2, 'never'],
+    // 'react/forbid-component-props': 2, TODO: we are discussing this
+    'react/forbid-foreign-prop-types': 2,
+    'react/forbid-prop-types': 1, // NOTE: will bump to error in a future major version
+    'react/jsx-boolean-value': 2,
+    'react/jsx-closing-bracket-location': 2,
+    'react/jsx-curly-spacing': 2,
+    'react/jsx-equals-spacing': 2,
+    'react/jsx-first-prop-new-line': [2, { multiline: true }],
+    'react/jsx-indent': [2, 2],
     'react/jsx-indent-props': [2, 2],
-    'react/jsx-max-props-per-line': 0,
+    'react/jsx-max-props-per-line': [2, { maximum: 1, when: 'multiline' }],
+    'react/jsx-no-bind': 2,
+    'react/jsx-no-comment-textnodes': 2,
     'react/jsx-no-duplicate-props': [2, { ignoreCase: true }],
-    'react/jsx-no-literals': 0,
-    'react/jsx-pascal-case': 2,
+    'react/jsx-no-target-blank': 2,
+    'react/jsx-pascal-case': [2, { allowAllCaps: true }],
+    'react/jsx-sort-props': 2,
+    'react/jsx-space-before-closing': [2, 'always'],
+    'react/jsx-tag-spacing': [2, {
+      afterOpening: 'never',
+      beforeSelfClosing: 'always',
+      closingSlash: 'never',
+    }],
     'react/jsx-wrap-multilines': 2,
+    'react/no-children-prop': 2,
+    'react/no-danger': 1,
+    'react/no-danger-with-children': 2,
     'react/no-did-mount-set-state': 1,
     'react/no-did-update-set-state': 1,
     'react/no-multi-comp': 1,
-    'react/self-closing-comp': 2,
+    'react/no-string-refs': 2,
+    'react/no-unescaped-entities': 2,
+    'react/no-unused-prop-types': 2,
+    'react/require-default-props': 2,
+    'react/self-closing-comp': [2, { component: true, html: true }],
     'react/sort-comp': 2,
-  }
+    'react/sort-prop-types': 2,
+    'react/style-prop-object': 2,
+    'react/void-dom-elements-no-children': 2,
```

Comment:
We definitely do in brigade-web [(through overcommit)](https://github.com/brigade/brigade-web/blob/51c09805a23516211d6deb818ddba94df126a66d/.overcommit.yml#L41), and I suggest we do it in other projects as well (unless we're already doing that).